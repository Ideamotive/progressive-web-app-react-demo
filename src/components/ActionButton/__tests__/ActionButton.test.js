import ActiveButton from '../ActionButton';
import React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';

describe('<ActionButton/> component', () => {
  test('should match to snapshot', () => {
    const tree = renderer
      .create(<ActiveButton/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render icon with class', () => {
    const iconClass = 'test';
    const wrapper = mount(<ActiveButton iconClass={iconClass}/>);
    expect(wrapper.find(`i.${iconClass}`)).toHaveLength(1);
  });

});

