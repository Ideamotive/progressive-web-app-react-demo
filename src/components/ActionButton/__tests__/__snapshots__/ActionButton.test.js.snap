// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton/> component should match to snapshot 1`] = `
<button
  className="sc-bdVaJa hOICsF"
  height="70px"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  width="70px"
>
  <i
    className="fa fa-eye sc-bwzfXH cpZBPd"
  />
</button>
`;
